apiserver/facades/agent/credentialvalidator/register.go

```diff
@@ -19,10 +20,22 @@ func Register(registry facade.FacadeRegistry) {
 
 // newCredentialValidatorAPI creates a new CredentialValidator API endpoint on server-side.
 func newCredentialValidatorAPI(ctx facade.ModelContext) (*CredentialValidatorAPI, error) {
+
+	modelCredentialWatcherGetter := func(stdCtx context.Context) (watcher.NotifyWatcher, error) {
+		return ctx.DomainServices().Model().WatchModelCloudCredential(stdCtx, ctx.ModelUUID())
+	}
+
+	domainServices := ctx.DomainServices()
```

Comment:
move this up higher and use it in the getter func

Reply:
Resolved with thanks ian!